### ISSUE TITLE
(OLD) V1.X.X Arduino-IDE freezes on compilation, (possibly) due to the amount of preproccesor directives

### ISSUE THREAD

**Issue by GiorgosXou** [GiorgosXou/NeuralNetworks#31]
**Depending on the board or MCU** you use, the IDE will either **freeze or** take way too long to **compile**. *(At least with [this build of the old-Arduino](https://archlinux.org/packages/extra/x86_64/arduino/))*. I'm suspecting this is a result of the immense amount of preproccesor-directives\optimizations in a single file. This issue appeared with the release of [NeuralNetworks v2.1.0-beta](https://github.com/GiorgosXou/NeuralNetworks/tree/v2.1.0-beta)

**PS.** I'm also suspecting this is a threading issue of the old IDE or something... because it compiles perfectly with `arduino-cli` or the new one

**tl;dr** Use the new one or `arduino-cli`

***Any Idea though?***

**Comment by GiorgosXou** [comment]
ha... this doesn't happen on my other Arch-btw system, lol
